src/Interpreters/InsertDependenciesBuilder.cpp

```diff
@@ -1125,7 +1142,7 @@ bool InsertDependenciesBuilder::observePath(const DependencyPath & path)
         views_error_registry->init(current);
 
         select_queries[current] = metadata->getSelectQuery().inner_query;
-        input_headers[current] = output_headers.at(path.parent(2));
```

Comment:
Are you sure here?

Reply:
Before that feature, there were always chains like `view->table->view->table`, so parent(2) always had a valid header.
For a chain like 'view->merge1->merge2->table` merge2 header may be different from merge1, so we need parent(1) here